web_generator/README.md

```diff
@@ -1,21 +1,133 @@
+definitions:
+
+#### Top Level Declarations
+- Top level declarations are handled as top level dart `external` declarations annotated with `@JS`
+
+#### Enums
+- Enums are represented as extension types with static members
+- For most cases, enums are represented without any external calls, and can have their rep type as Dart primitive types where possible
+- If the value for an enum is not given, then the whole enum is treated as an external interop member, and any members without a value are marked as `external`
+
+#### Interfaces and Classes
+- Interfaces and Classes are emitted as extension types that wrap and implement `JSObject`
+- Interface and Class inheritance is maintained using `implements` between extension types
+- Classes have default constructors if none are provided
+- Interface and Classes support members such as: properties, functions, operators, `call` declarations, construct signatures, getters and setters.
+- Readonly properties are represented as getters.
+- Overriding signatures are annotated with [`@redeclare`](https://pub.dev/documentation/meta/latest/meta/redeclare-constant.html). 
+- Multiple instances of interfaces in a given scope are merged together into a single interface. Also supports merging with `var` declarations.
+
+#### Namespaces
+- Namespaces in TS are converted to extension types on `JSObject` with static members.
+- Typed declarations (such as enums, classes, interfaces and namespaces) are prefixed and generated in the general module space.
+- Classes are generated as well as static methods that redirect to constructor calls.
+- Namespaces nested inside the namespace are as well generated as static getters on the
+- Supports overloading and declaration merging with classes, interfaces, enums, `var` declarations, and functions.
+
+#### Types
+- Supports mapping basic JS types to Dart `js_interop` types.
+- `never` type returned from a const declaration, function, readonly property or method has the declaration annotated with [`@doNotStore`](https://pub.dev/documentation/meta/latest/meta/doNotStore-constant.html). 
+- Supports automatically mapping web types to `package:web` types
+- Anonymous Unions/Intersections: Anonymous unions and intersections are represented as extension types with cast members to cast the value as a type in the union. If the union is homogenous and contains strings/numbers, then it is generated more like an enum, with specific static values. An intersection with `null` or `undefined` equals `never`
+- Anonymous Objects: Anonymous objects are represented similar to interface declarations, with an object literal constructor.
+- Anonymous Closures and Construct Signatures: Anonymous closures and construct signatures are represented as extension types on `JSFunction` overriding the `call` method. This allows for more typing as opposed to just `JSFunction`. While closures have their calls as `external`, construct signatures were redirect calls to the class constructor itself.
```

Comment:
"This allows for a stronger type signature than just `JSFunction`. While closures..."